### PR TITLE
[ROCKETMQ-33] CPU Occupy 100%

### PR DESCRIPTION
**Pull request by qinliujie**
https://issues.apache.org/jira/browse/ROCKETMQ-33

### CODE REVIEW

**Comment by qinliujie** [comment]
Sorry,I wrote the wrong issue number,it shoud be [ROCKETMQ-33] CPU Occupy 100%

File: store/src/main/java/org/apache/rocketmq/store/CommitLog.java [hunk 1]
```diff
@@ -1045,7 +1045,7 @@ public void run() {
 
             while (!this.isStopped()) {
                 try {
-                    this.waitForRunning(0);
+                    this.waitForRunning(1000);
```
Comment (zhouxinyu) [inline store/src/main/java/org/apache/rocketmq/store/CommitLog.java]: `GroupCommitService` will be used when broker is in `SYNC_FLUSH`, so the flushCommitLogService need always `doCommit` without a break.

And this won't cause 100% CPU usage unless your server only has one cpu core.

BTW, how about use `ASYNC_FLUSH` if you mind.

Comment (zhouxinyu) [inline store/src/main/java/org/apache/rocketmq/store/CommitLog.java]: I also think `waitForRunning(0)` is too rigorous, but `waitForRunning(1000)` is adventurous while the `GroupCommitService` may miss some notify signals.

How about set a smaller wait time, like 10ms? what's your opinion ? @qinliujie

**Comment by qinliujie** [comment]
When I use ASYNC_MASTER,there was the same problem,Because  HAService also has `this.waitForRunning(0)`,whether SYNC_MASTER or not.

When you said that the `GroupCommitService` may miss some notify signals,I think it's right.So I think wait time to be 10ms is acceptable.

**Comment by vintagewang** [comment]
review ok.

**Comment by zhouxinyu** [comment]
Hi, @qinliujie  This PR will be merged soon, but next time please send PR from branch ROCKETMQ-33 instead of master.

**Comment by qinliujie** [comment]
@zhouxinyu OK.

**Comment by Ah39** [comment]
waitForRunning(0) is call wait(0). wait(0)  is wait forever util somenoe notify the Thread

**Comment by Jaskey** [comment]
I agreee with @Ah39  , 0ms is wait forever, which should not cause high cpu problem, any other justice reason to modify it to 10ms?

**Comment by qinliujie** [comment]
old version use the Object wait() method,but it had changed to countdown latch in new version,please notice that @Jaskey

**Pull request closed without merging**